### PR TITLE
Feature: Apply design changes to Select.tsx

### PR DESCRIPTION
**Pull request by AnukratiMehta**
## Overview

This pull request addresses the issue #127 by updating the `SelectContent`, `SelectLabel`, and `SelectItem` components to adhere to the design specifications provided in Figma.

## Changes

- Replaced the caret icon with a new `CheckIcon` as per the Figma design.
- Adjusted padding, border radius, font size, and font weight per Figma design guidelines.

## Screen Captures
Screenshot 2024-02-22 at 1 48 29 AM

## Assignee Checklist:

- [x] The naming convention of the PR title is correct (See the comment at the top of this template)
- [x] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [x] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [x] The correct assignees and reviewers have been designated for this PR
- [x] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] All the related issues are associated with this PR
- [x] All criteria in the associated issues are met (please tick the checkboxes)
- [x] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [x] Double-check the "Assignee Checklist"
- [x] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] The code is well-commented and easy to understand
- [x] The UI changes accurately reflect the provided design

### CODE REVIEW

File: src/components/ui/Select/Select.tsx [hunk 1]
```diff
@@ -97,7 +97,7 @@ const SelectItem = forwardRef<
     <span className="absolute left-2">
       <PrimitiveItemIndicator>
         {/* TODO:Create a new icon to use in here */}
-        <Icon icon={CaretDownIcon} size={4} />
+        <Icon icon={CheckIcon} size={4} className="text-primary" />
```
Comment (nick-y-ito) [inline src/components/ui/Select/Select.tsx:100]: The size of the icon itself has to be `3.5` to be `14px`. Maybe you

image

You could add a wrapper element to make it a 16*16px square.

image
image

File: src/components/ui/Select/Select.tsx [hunk 2]
```diff
@@ -87,7 +87,7 @@ const SelectItem = forwardRef<
     ref={ref}
     className={cn(
       'relative flex items-center',
-      'w-full rounded-sm text-sm outline-none  focus:bg-gray-light',
+      'w-full rounded-sm text-base font-normal outline-none focus:bg-gray-light',
```
Comment (nick-y-ito) [inline src/components/ui/Select/Select.tsx:90]: Could you please add the class `data-[state=checked]:bg-gray-light` to highlight the selected item?
The usage of this class is mentioned here:
https://www.radix-ui.com/primitives/docs/components/select#item

As the `className` may get longer, please consider the style mentioned here:
https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md#css-class-management

Comment (nick-y-ito) [reply src/components/ui/Select/Select.tsx:90]: The select items should also have a 6px of border-radius (= `rounded-md`), as per the Figma desing.
image

**Review by nick-y-ito — CHANGES_REQUESTED** [review]
@AnukratiMehta 
Thank you for your excellent work!

While the design is largely acceptable, I've added some comments to enhance its accuracy. Could you please take a moment to review them?

File: src/components/ui/Select/Select.tsx (hunk 2 above)
Comment (AnukratiMehta) [reply src/components/ui/Select/Select.tsx]: Thanks, I've grouped the classes based on:
- Position and alignment
- Size and shape
- Padding and cursor
- Colors

**Comment by nick-y-ito** [comment]
@AnukratiMehta 

Are you still in progress? If you are done with your change, please re-mention us to ask for a second review.

Currently, some errors are happening in the GitHub Actions. I think they will be solved if you merge the `develop` branch.

**Comment by AnukratiMehta** [comment]
@nick-y-ito 

No, the changes are done. The checks are passing too now. Hope this works!

File: src/components/ui/Select/Select.tsx [hunk 3]
```diff
@@ -86,18 +86,20 @@ const SelectItem = forwardRef<
...
+      'focus:bg-gray-light data-[state=checked]:bg-gray-light',
       className,
     )}
     {...props}
   >
     <span className="absolute left-2">
       <PrimitiveItemIndicator>
         {/* TODO:Create a new icon to use in here */}
-        <Icon icon={CaretDownIcon} size={4} />
+        <div className="w-4 h-4 flex items-center">
```
Comment (nick-y-ito) [inline src/components/ui/Select/Select.tsx:100]: To alight to center, you need both `items-center` and `justify-center`

**Review by nick-y-ito — CHANGES_REQUESTED** [review]
@AnukratiMehta 
Thank you for the update! Overall, it looks great! There is only one comment I left, which I hope you can kindly take a look at.

However, I might fix it and merge this PR shortly because there are a few conflicts to be resolved quickly.

**Comment by nick-y-ito** [comment]
@AnukratiMehta 
Since there was a conflict with the base branch (`develop`), I resolved it in the newly created branch `feature/127-apply-design-to-SelectContent-resolve-conflict` and also fixed the issues I pointed out in the last comment.

Please utilize it as needed. I think you can merge the branch into yours or cherry-pick the commits.

**Comment by AnukratiMehta** [comment]
> @AnukratiMehta Since there was a conflict with the base branch (`develop`), I resolved it in the newly created branch [`feature/127-apply-design-to-SelectContent-resolve-conflict`](https://github.com/nishiki-tech/nishiki-frontend/tree/feature/127-apply-design-to-SelectContent-resolve-conflict) and also fixed the issues I pointed out in the last comment.
> 
> Please utilize it as needed. I think you can merge the branch into yours or cherry-pick the commits.

@nick-y-ito 

Thanks. I have some questions. I've messaged you. Could you please check? I think it'll be quicker and easier to address them there :)

**Pull request merged**